### PR TITLE
feat(cmd|node): MutualPeers Node option and CLI flag

### PR DESCRIPTION
**Pull request by Wondertan**
Quick PR to add configurability for mutual peers from CLI

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -73,6 +74,16 @@ func Start(repoFlagName string, tp node.Type) *cobra.Command {
...
+			mutualPeersStr := cmd.Flag(mutualPeersFlag.Name).Value.String()
+			if mutualPeersStr != "" {
+				mutualPeers := strings.Split(mutualPeersStr, ",")
+				if len(mutualPeers) == 0 {
+					return fmt.Errorf("%s flag is passed but no address were given", mutualPeersFlag.Name)
+				}
+
+				opts = append(opts, node.WithMutualPeers(mutualPeers))
+			}
+
```
Comment (liamsi) [inline cmd/start.go:86]: Q: Should this be refactored into a common function?

**Review by liamsi — DISMISSED** [review]
LGTM

File: cmd/flags.go [hunk 2]
```diff
@@ -37,4 +38,10 @@ var (
 			"Example: <protocol>://<ip>:<port>, tcp://127.0.0.1:26657",
 		DefValue: "",
 	}
+	mutualPeersFlag = &flag.Flag{
+		Name: "p2p.mutual",
+		Usage: "Comma-Multiaddresses of mutual peers to keep unbreakable connection with. " +
+			"Peers must bidirectionally point to each other.",
```
[the remark this answers is not in the document]
Comment (Wondertan) [reply cmd/flags.go:44]: Nothing

File: cmd/start.go (hunk 1 above)
Comment (Wondertan) [reply cmd/start.go:86]: Yeah, eventually.

File: cmd/flags.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (liamsi) [reply cmd/flags.go:44]: Nothing, as in the they do not connect to each other?

[the remark this answers is not in the document]
Comment (Wondertan) [reply cmd/flags.go:44]: Yes

File: cmd/start.go (hunk 1 above)
Comment (renaynay) [reply cmd/start.go:86]: can u do in this PR? just make it a common flag the way the other common flags were declared.

Comment (renaynay) [reply cmd/start.go:86]: Otherwise we'd never get to it 😅

Comment (Wondertan) [reply cmd/start.go:86]: > can u do in this PR?
Yesss

**Comment by Wondertan** [comment]
re__based__

File: cmd/init.go [hunk 3]
```diff
@@ -59,6 +60,16 @@ func Init(repoFlagName string, tp node.Type) *cobra.Command {
 				opts = append(opts, node.WithRemoteCore(protocol, ip))
 			}
 
+			mutualPeersStr := cmd.Flag(mutualPeersFlag.Name).Value.String()
+			if mutualPeersStr != "" {
+				mutualPeers := strings.Split(mutualPeersStr, ",")
+				if len(mutualPeers) == 0 {
+					return fmt.Errorf("%s flag is passed but no address were given", mutualPeersFlag.Name)
```
Comment (adlerjohn) [inline cmd/init.go:67]:
```suggestion
					return fmt.Errorf("%s flag is passed but no addresses were given", mutualPeersFlag.Name)
```

File: cmd/flags.go [hunk 4]
```diff
@@ -37,4 +38,10 @@ var (
 			"Example: <protocol>://<ip>:<port>, tcp://127.0.0.1:26657",
 		DefValue: "",
 	}
+	mutualPeersFlag = &flag.Flag{
+		Name: "p2p.mutual",
+		Usage: "Comma-separated Multiaddresses of mutual peers to keep unbreakable connection with. " +
```
Comment (adlerjohn) [inline cmd/flags.go:43]: Is the multiaddress format documented somewhere?
```suggestion
		Usage: "Comma-separated multiaddresses of mutual peers to keep an unbreakable connection with. " +
```

File: cmd/start.go [hunk 5]
```diff
@@ -73,6 +74,16 @@ func Start(repoFlagName string, tp node.Type) *cobra.Command {
 				opts = append(opts, node.WithConfig(cfg))
 			}
 
+			mutualPeersStr := cmd.Flag(mutualPeersFlag.Name).Value.String()
+			if mutualPeersStr != "" {
+				mutualPeers := strings.Split(mutualPeersStr, ",")
+				if len(mutualPeers) == 0 {
+					return fmt.Errorf("%s flag is passed but no address were given", mutualPeersFlag.Name)
```
Comment (adlerjohn) [inline cmd/start.go:81]:
```suggestion
					return fmt.Errorf("%s flag is passed but no addresses were given", mutualPeersFlag.Name)
```

File: cmd/flags.go (hunk 4 above)
Comment (Wondertan) [reply cmd/flags.go:43]: https://multiformats.io/multiaddr/

Comment (Wondertan) [reply cmd/flags.go:43]: or http://multiformats.io.ipns.localhost:8080/multiaddr/ for cool people ;)

Comment (adlerjohn) [reply cmd/flags.go:43]: Is it possible to put a link to that in the docstring?

Comment (Wondertan) [reply cmd/flags.go:43]: I think so. This is a string that is shown to you once you do `--help`.

**Comment by Wondertan** [comment]
@adlerjohn, added links to multiformats.io/multiaddr

**Review by adlerjohn — DISMISSED** [review]

**Comment by Wondertan** [comment]
@adlerjohn, I put the changelog entry into the wrong category and need your approval again 😅

**Review by adlerjohn — DISMISSED** [review]

**Review by Bidon15 — DISMISSED** [review]

File: cmd/flags.go [hunk 6]
```diff
@@ -37,4 +38,10 @@ var (
 			"Example: <protocol>://<ip>:<port>, tcp://127.0.0.1:26657",
 		DefValue: "",
 	}
+	mutualPeersFlag = &flag.Flag{
+		Name: "p2p.mutual",
+		Usage: "Comma-separated multiaddresses of mutual peers to keep an unbreakable connection with. " +
```
Comment (renaynay) [inline cmd/flags.go:43]: "unbreakable connection" isn't really that descriptive, does it just mean that the peers re-dial each other if the connection breaks? Maybe mention that.

**Review by renaynay — DISMISSED** [review]
One comment, but otherwise LGTM

File: cmd/flags.go (hunk 6 above)
Comment (Wondertan) [reply cmd/flags.go:43]: Unbreakable means it is immune to peer scoring slashing and trimming by connection manager, I.e. Prioritized. Sorry for the confusion. 

So your guess is a bit wrong, but still valid as they try to reconnect to each other if having troubles

Comment (adlerjohn) [reply cmd/flags.go:43]: If the reader guesses wrong, that's indicative of a need for more docs! Can you improve the wording here to include what you just wrote @Wondertan?

Comment (Wondertan) [reply cmd/flags.go:43]: True, adding more info.

**Review by adlerjohn — APPROVED** [review]

File: CHANGELOG-PENDING.md [hunk 7]
```diff
@@ -6,12 +6,10 @@ Month, DD, YYYY
...
 ### BREAKING CHANGES
 
-- [go package] (Link to PR) Description @username
-[feat(cmd): give a birth to cel-shed and p2p key utilities #281](https://github.com/celestiaorg/celestia-node/pull/281) [@Wondertan](https://github.com/Wondertan)
-
 ### FEATURES
 
-- [go package] (Link to PR) Description @username
+- [feat(cmd): give a birth to cel-shed and p2p key utilities #281](https://github.com/celestiaorg/celestia-node/pull/281) [@Wondertan](https://github.com/Wondertan)
+- [feat(cmd|node): MutualPeers Node option and CLI flag #280](https://github.com/celestiaorg/celestia-node/pull/280) [@Wondertan](https://github.com/Wondertan)
```
Comment (Bidon15) [inline CHANGELOG-PENDING.md:12]: I suggest it should look smt like this: 
```suggestion
- [cmd](https://github.com/celestiaorg/celestia-node/pull/281) feat(cmd): give a birth to cel-shed and p2p key utilities #281 @Wondertan
- [cmd|node](https://github.com/celestiaorg/celestia-node/pull/280) feat(cmd|node): MutualPeers Node option and CLI flag @Wondertan
```

Comment (Wondertan) [reply CHANGELOG-PENDING.md:12]: Nope

Comment (Wondertan) [reply CHANGELOG-PENDING.md:12]: `Cmd` is a package and you are making a link to PR from the package word that doesn't make sense
Also, you remove a link to contributor

Comment (Bidon15) [reply CHANGELOG-PENDING.md:12]: Why? I was referring to Josh's example below in bug fixes section. 
Quote:
```
- [header] Added missing `err` value in ErrorW logging calls. @jbowen93
```

Comment (Bidon15) [reply CHANGELOG-PENDING.md:12]: I just want to clarify this for myself, as I've got confused when editing changelog-pending.md file for my race action PR 😅

File: CHANGELOG-PENDING.md [hunk 8]
```diff
@@ -6,12 +6,10 @@ Month, DD, YYYY
 
 ### BREAKING CHANGES
 
-- [go package] (Link to PR) Description @username
-[feat(cmd): give a birth to cel-shed and p2p key utilities #281](https://github.com/celestiaorg/celestia-node/pull/281) [@Wondertan](https://github.com/Wondertan)
-
 ### FEATURES
 
-- [go package] (Link to PR) Description @username
+- [feat(cmd): give a birth to cel-shed and p2p key utilities #281](https://github.com/celestiaorg/celestia-node/pull/281) [@Wondertan](https://github.com/Wondertan)
```
Comment (renaynay) [inline CHANGELOG-PENDING.md:11]: Wait why is this still in the PR?

**Review by renaynay — APPROVED** [review]

**Pull request merged**